### PR TITLE
Add Popsicle

### PR DESCRIPTION
**Pull request by mmstick**
Closes https://github.com/pop-os/popsicle/issues/43

### CODE REVIEW

**Comment by mmstick** [comment]
@cassidyjames Truth be told, I've no idea how all the appstream stuff works on Debian. What would be the best course of action for getting appstream data working with the Pop!_Shop?

**Comment by brs17** [comment]
It shows up now! I'm thinking we probably should include a screenshot though.

Are we happy with the description? (@cassidyjames )

screenshot from 2018-06-26 15-10-56

**Comment by mmstick** [comment]
I think we should have screenshots of each view, as well as a better description.

**Comment by cassidyjames** [comment]
Screenshots in appstream data shipped this was is apparently a pain. @btkostner says there's some work we can do, but it involves a lot more than just sticking a URL in the data.

File: src/data/popsicle.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - name: popsicle.png
+      width: 64
+      height: 64
+    - name: popsicle.png
+      width: 128
+      height: 128
+Screenshots:
+  - thumbnails: []
+    source-image:
+    url: https://raw.githubusercontent.com/pop-os/popsicle/master/demo.gif
```
Comment (cassidyjames) [inline src/data/popsicle.yml]: afaik GIFs aren't supported in AppStream

Comment (cassidyjames) [reply src/data/popsicle.yml]: Yeah, just PNG or JPG https://www.freedesktop.org/software/appstream/docs/sect-Metadata-Application.html#tag-dapp-screenshots

**Comment by cassidyjames** [comment]
I've taken a bunch of HiDPI screenshots with shadows and everything for whenever we figure out where to put them. I've also updated the description.

File: dest/data/popsicle.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      width: 64
+      height: 64
+    - name: popsicle.png
+      width: 128
+      height: 128
+Screenshots:
+  - thumbnails: []
+    source-image:
+    url: https://raw.githubusercontent.com/pop-os/popsicle/master/demo.gif
+    lang: C
```
Comment (jackpot51) [inline dest/data/popsicle.yml]: Should be Rust

File: dest/data/popsicle.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - name: popsicle.png
+      width: 64
+      height: 64
+    - name: popsicle.png
+      width: 128
+      height: 128
+Screenshots:
+  - thumbnails: []
+    source-image:
+    url: https://raw.githubusercontent.com/pop-os/popsicle/master/demo.gif
```
Comment (jackpot51) [inline dest/data/popsicle.yml]: Please use PNG

File: src/data/popsicle.yml (hunk 1 above)
Comment (jackpot51) [reply src/data/popsicle.yml]: Please use PNG

File: src/data/popsicle.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    lang: C
```
Comment (jackpot51) [inline src/data/popsicle.yml]: Should be Rust

**Comment by jackpot51** [comment]
Can you run build so the dest folder is updated? Then I think this is ready

**Comment by jackpot51** [comment]
@brs17 thoughts?

**Comment by brs17** [comment]
Do we want to add a "Report a Problem" link that would point to https://github.com/pop-os/popsicle/issues ?

The screenshots appear to be a little lower resolution than is probably ideal.

Also in the description of the application has the html visible:

```
<p>Write an ISO or other image to multiple USB devices all at once. Easily preparing a bunch of flash drives of your favorite OS with just a couple of clicks.</p> <ul>
  <li>Supports USB 2 and 3 devices</li>
  <li>Use USB hubs for massively parallel writing</li>
  <li>Verify your image with the SHA256 or MD5 checksum</li>
  <li>Check the progress, speed, and success of each device while flashing</li>
  <li>Open ISO or IMG files from the app, or straight from your file manager</li>
<ul>
```

**Comment by brs17** [comment]
This is a screenshot of what I see on the current commit (bdf2db):

popsicle

Somehow doesn't seem to include the fixes for the issues I mentioned earlier. Those fixes should add the issues page link as well as remove the tags from the description.

In regards to the screenshots, I still think they look a little blurry and I am wondering if those screenshots are taken on a hidpi screen if they would appear crisper.

**Comment by mmstick** [comment]
Seems I edited the wrong file.

**Comment by cassidyjames** [comment]
You can take the screenshots on a HiDPI display, or simply use the GTK inspector to force 2x scaling. In the past, I also used the [elementary screenshot tool](https://github.com/elementary/screenshot) for higher-quality screenshots that include the window shadow exactly as defined by the GTK stylesheet as well.

**Comment by isantop** [comment]
popsicle-hidpi_screenshot_appdata

HiDPI screenshot

**Comment by brs17** [comment]
@mmstick , looks like in your latest change we lost the icon for popsicle.

screenshot from 2018-10-12 12-07-30
screenshot from 2018-10-12 12-07-43


It also looks like the `Help` button links to the issues page. I would expect `Report a Problem` to link there. Not sure we have a good page to link `Help` to right now.
screenshot from 2018-10-12 12-07-38

**Comment by brs17** [comment]
Not going to worry about improved screenshots here as that should be a pr into Popsicle itself.

**Comment by brs17** [comment]
With that latest commit, I believe we can/should merge this. @mmstick can you confirm the links at the bottom of the popsicle page work as expected?

**Comment by mmstick** [comment]
Sure. I'll check it out in the morning.

**Pull request merged**